dashboard/test/ui/features/step_definitions/sections.rb

```diff
@@ -0,0 +1,129 @@
+And /^I create a new section$/ do
+  individual_steps %Q{
+    When I press the new section button
+    Then I should see the new section dialog
+
+    When I select email login
+    And I press the save button to create a new section
+    And I wait for the dialog to close
+    Then I should see the section table
+  }
+end
+
+And /^I create a new section with course "([^"]*)"(?: and unit "([^"]*)")?$/ do |primary, secondary|
+  individual_steps %Q{
+    When I press the new section button
+    Then I should see the new section dialog
+
+    When I select email login
+    Then I wait to see "#uitest-primary-assignment"
+
+    When I select the "#{primary}" option in dropdown "uitest-primary-assignment"
+  }
+
+  if secondary
+    individual_steps %Q{
+      And I wait to see "#uitest-secondary-assignment"
+      And I select the "#{secondary}" option in dropdown "uitest-secondary-assignment"
+    }
+  end
+
+  individual_steps %Q{
+    And I press the save button to create a new section
+    And I wait for the dialog to close
+    Then I should see the section table
+  }
+end
+
+And(/^I save the section url$/) do
+  section_code = @browser.execute_script <<-SCRIPT
+    return document
+      .querySelector('.uitest-owned-sections tbody tr:last-of-type td:nth-child(6)')
+      .textContent
+      .trim();
+  SCRIPT
+  @section_url = "http://studio.code.org/join/#{section_code}"
+end
+
+And(/^I navigate to the section url$/) do
+  steps %Q{
+    Given I am on "#{@section_url}"
+  }
+  wait_short_until {/^\/join/.match(@browser.execute_script("return location.pathname"))}
+end
+
+And(/I type the section code into "([^"]*)"$/) do |selector|
+  puts @section_url
+  section_code = @section_url.split('/').last
+  steps %Q{
+    And I type "#{section_code}" into "#{selector}"
+  }
+end
+
+When /^I see the section set up box$/ do
+  steps 'When I wait to see ".uitest-set-up-sections"'
+end
+
+When /^I press the new section button$/ do
+  steps 'When I press the first ".uitest-newsection" element'
+end
+
+Then /^I should see the new section dialog$/ do
+  steps 'Then I see ".modal"'
```

Comment:
There are multiple steps here that wrap another single step, presumably so the tests are more human readable?  For next time, is that preferable to using the step that already exists? e.g. when writing the feature instead of "Then I should see the new section dialog" why not just use "Then I see '.modal'"?

Reply:
Good question :-) a one-line "method" like this is a valid form of self-documenting code.

If your test is still readable substituting `Then I see ".modal"` then that's a bit better because you can see more clearly what steps are being executed under the hood. But if you end up needing to comment that line to say what modal it is looking for, then it seems like either adding a short comment or extracting a 1-line method are both fine options and you can choose based on personal preference.